src/ApolloNetworkStatusProvider.tsx

```diff
@@ -1,23 +1,24 @@
-import React, {useMemo, ReactNode} from 'react';
-import {withApollo, ApolloProvider} from 'react-apollo';
-import ApolloClient from 'apollo-client';
+import {ApolloProvider, getApolloContext} from '@apollo/react-common';
+import React, {useMemo, ReactNode, useContext} from 'react';
 import Dispatcher from './Dispatcher';
 import ApolloNetworkStatusDispatcherContext from './ApolloNetworkStatusDispatcherContext';
 import augmentApolloClient from './augmentApolloClient';
 
 type Props = {
-  client: ApolloClient<any>;
   children: ReactNode;
   enableBubbling?: boolean;
 };
 
-function ApolloNetworkStatusProvider({
-  client,
-  children,
-  enableBubbling
-}: Props) {
+function ApolloNetworkStatusProvider({children, enableBubbling}: Props) {
+  const {client} = useContext(getApolloContext());
   const dispatcher = useMemo(() => new Dispatcher(), []);
 
+  if (!client) {
+    throw new Error(
```

Comment:
We'll probably never get to here when the context isn't defined. We should do this check before we destructure the `client`.

Reply:
Actually no. TypeScript is right, `getApolloContext` returns at least an empty object, but the `client` is optional.